eth_account/_utils/structured_data/hashing.py

```diff
@@ -39,15 +37,21 @@ def get_dependencies(primary_type, types):
         deps.add(struct_name)
         fields = types[struct_name]
         for field in fields:
-            if field["type"] not in types:
+            field_type = field["type"]
+
+            # Handle array types
+            if is_array_type(field_type):
+                field_type = field_type[:field_type.index('[')]
+
+            if field_type not in types:
                 # We don't need to expand types that are not user defined (customized)
                 continue
-            elif field["type"] in deps:
+            elif field_type in deps:
                 # skip types that we have already encountered
                 continue
             else:
```

Comment:
Since one of the bugs was that we were skipping over an array where we didn't know what it was, I wonder if we should make this an `elif` and raise on the else? Or if we could even define that condition?